Actions/DetermineArtifactUrl/DetermineArtifactUrl.ps1

```diff
@@ -22,40 +21,31 @@ try {
     
     #region Action: Determine artifacts to use
     $telemetryScope = CreateScope -eventId 'DO0084' -parentTelemetryScopeJson $parentTelemetryScopeJson
-    $secrets = $secretsJson | ConvertFrom-Json | ConvertTo-HashTable
-    if ($secrets.ContainsKey('insiderSasToken')) {
-        $insiderSasToken = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets.insiderSasToken))
-    } else {
-        $insiderSasToken = ""
-    }
-    $projectSettings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable
-    $projectSettings = AnalyzeRepo -settings $projectSettings -project $project -doNotCheckArtifactSetting -doNotIssueWarnings
-    $artifactUrl = Determine-ArtifactUrl -projectSettings $projectSettings -insiderSasToken $insiderSasToken
+    $secrets = $env:Secrets | ConvertFrom-Json | ConvertTo-HashTable
+    $insiderSasToken = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($Secrets.insiderSasToken))
+    $settings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable
+    $settings = AnalyzeRepo -settings $settings -project $project -doNotCheckArtifactSetting -doNotCheckAppDependencyProbingPaths -doNotIssueWarnings
+    $artifactUrl = Determine-ArtifactUrl -projectSettings $settings -insiderSasToken $insiderSasToken
```

Comment:
`AnalyzeRepo` is already calling `Determine-ArtifactUrl` if the `doNotCheckArtifactSetting` switch isn't set.

Reply:
Yes, but AnalyzeRepo will also download the artifacts - which it shouldn't.
All-in-all - AnalyzeRepo needs to go away and have functions like Determine-ArtifactUrl to take over